index.js

```diff
@@ -1091,27 +1103,52 @@
           this._registerTrackers();
         }
 
-        // activates page tracking
-        if (trackRoutes) {
-          $rootScope.$on(pageEvent, function () {
-            // Apply $route based filtering if configured
-            if (readFromRoute) {
+        // This section adds different listeners based on the configuration
+        var eventListener = null;
+        if (typeof trackRouteFn === 'function') {
+          eventListener = function() {
+            // Redirect event to custom function, passing in protected object
+            trackRouteFn.apply(that, arguments);
+          };
+        } else if (trackRoutes) {
+          if (hybridMobileSupport) {
+            // For mobile devices we need to track a screen rather than a page
+            eventListener = function (event, state) {
+              that._send('screenview', {
```

Comment:
Log the warning inside the eventListener function, if you weren't already thinking of doing that. Do you think it should still make the _send call?